### PR TITLE
Replace devcert with selfsigned

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Replace `devcert` with `selfsigned`, which is already installed for webpack dev server.
- Update the Express server to use a self-signed certificate in development.
- Remove `start:dev` and `dev-server.js`.
- Update the app READMEs.

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
app-content-pages
app-project

## Linked Issue and/or Talk Post
- #4103 

## How to Review
Start a dev server as per usual or run either NextJS app on `local.zooniverse.org`. 

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## Maintenance
- [x] If not from dependabot, the PR creator has described the update (major, minor, or patch version, changelog)

### CODE REVIEW

File: packages/app-content-pages/server/server.js [hunk 1]
```diff
@@ -4,15 +4,21 @@ if (process.env.NEWRELIC_LICENSE_KEY) {
...
 const setCacheHeaders = require('./set-cache-headers')
 
 const port = parseInt(process.env.PORT, 10) || 3000
+const hostname = 'local.zooniverse.org'
 const dev = process.env.NODE_ENV !== 'production'
 const app = next({ dev })
 const handle = app.getRequestHandler()
 
+const key = fs.readFileSync('server.key');
+const cert = fs.readFileSync('server.cert');
```
Comment (eatyourgreens) [inline packages/app-content-pages/server/server.js:20]: This crashes if either `server.cert` or `server.key` don't exist, which they won't in a staging or production deploy. I need to read up some more on how Node filesystem commands work. 🤔

Comment (eatyourgreens) [reply packages/app-content-pages/server/server.js]: Fixed with `fs.existsSync(path)` for both files.

**Review by mcbouslog — APPROVED** [review]
Reviewed selfsigned on [GitHub](https://github.com/jfromaniello/selfsigned) and [npm](https://www.npmjs.com/package/selfsigned), compared to devcert [GitHub](https://github.com/davewasmer/devcert) and [npm](https://www.npmjs.com/package/devcert) - selfsigned appears better maintained and significantly more widely adopted 👍 . Instructions are straight-forward, great implementation!

Tested local environments for...

app-project:
- [x] `yarn dev` works as expected, loaded https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats (signed in from existing staging session)
- [x] `yarn build` then `yarn start` works as expected, loaded https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess?env=production&demo=true (signed in from existing production session)
- `docker-compose up -d` errored with "Error: Cannot find module 'selfsigned'"

app-content-pages:
- [x] `yarn dev` works as expected, loaded https://localhost.zooniverse.org:3000/about/publications (signed in from existing staging session)
- [x] `yarn build` then `yarn start` works as expected, loaded https://localhost.zooniverse.org:3000/about/team (signed in from existing production session)
- `docker-compose up` errored with "Error: Cannot find module 'selfsigned'"

Are the docker errors expected or likely something I'm doing wrong on my end with docker (which I rarely use locally)? I see the notes in the READMEs "Running in development," Docker sections that note `devcert` (which can be updated to `selfsigned`) isn't setup with Docker, so I think these errors are expected, but want to confirm.

**Comment by eatyourgreens** [comment]
The Docker errors sound like you might need to run `docker compose build`, which should rebuild the monorepo inside the image.

**Comment by eatyourgreens** [comment]
Let’s try this out on staging. 👍🏻

**Pull request merged**